modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/AsyncHttpClientClientGenerator.scala

```diff
@@ -0,0 +1,770 @@
+                new Parameter(util.EnumSet.of(FINAL), argType.containedType.unbox, new SimpleName(param.getNameAsString))
+              ).toNodeList).setBody(new BlockStmt(List(
+                methodCall,
+                new ReturnStmt(new ThisExpr)
+              ).toNodeList))
+            })
+          optionalParamMethods.foreach(callBuilderCls.addMember)
+
+          callBuilderCls.addMethod("withHeader", PUBLIC)
+            .setParameters(List(
+              new Parameter(util.EnumSet.of(FINAL), STRING_TYPE, new SimpleName("name")),
+              new Parameter(util.EnumSet.of(FINAL), STRING_TYPE, new SimpleName("value")),
+            ).toNodeList)
+            .setType(callBuilderType)
+            .setBody(new BlockStmt(List(
+              new ExpressionStmt(new MethodCallExpr(
+                new FieldAccessExpr(new ThisExpr, "builder"),
+                "addHeader",
+                List[Expression](new NameExpr("name"), new NameExpr("value")).toNodeList
+              )),
+              new ReturnStmt(new ThisExpr)
+            ).toNodeList))
+
+          val httpMethodCallExpr = new MethodCallExpr(
+            new FieldAccessExpr(new ThisExpr, "httpClient"),
+            "apply",
+            new NodeList[Expression](new MethodCallExpr(new NameExpr("builder"), "build"))
+          )
+          val requestCall = new MethodCallExpr(httpMethodCallExpr, "thenApply", new NodeList[Expression](
+            new LambdaExpr(new NodeList(new Parameter(util.EnumSet.of(FINAL), RESPONSE_TYPE, new SimpleName("response"))), new BlockStmt(new NodeList(
+              new SwitchStmt(new MethodCallExpr(new NameExpr("response"), "getStatusCode"), new NodeList(
+                responses.value.map(response => new SwitchEntryStmt(new IntegerLiteralExpr(response.statusCode), new NodeList(response.value match {
+                  case None => new ReturnStmt(new ObjectCreationExpr(null, JavaParser.parseClassOrInterfaceType(s"${responseParentName}.${response.statusCodeName.asString}"), new NodeList()))
+                  case Some((valueType, _)) => new TryStmt(
+                    new BlockStmt(new NodeList(
+                      new ExpressionStmt(new AssignExpr(
+                        new VariableDeclarationExpr(new VariableDeclarator(valueType, "result"), FINAL),
+                        new MethodCallExpr(
+                          new FieldAccessExpr(new ThisExpr, "objectMapper"),
+                          "readValue",
+                          new NodeList[Expression](
+                            new MethodCallExpr(new NameExpr("response"), "getResponseBodyAsStream"),
+                            jacksonTypeReferenceFor(valueType)
+                          )
+                        ), AssignExpr.Operator.ASSIGN)),
+                      new IfStmt(
+                        new BinaryExpr(new NameExpr("result"), new NullLiteralExpr, BinaryExpr.Operator.EQUALS),
```

Comment:
Unfortunately, it's inconsistent.  I've seen it return `null` when it can't deserialize properly, but also throw `JsonProcessingException`.

Reply:
I really hope this is not the case. I really want jackson to just throw an exception if it can't parse something, otherwise return the native object representation (deserializing a JSON file with the single `null` keyword should return a `null`)

Reply:
I've also seen cases where Jackson will just return an object with missing fields set to `null`, which is basically worse.  Need to look into this a bit more to check the behavior.